sdk/azidentity/managed_identity_client.go

```diff
@@ -171,14 +173,16 @@ func (c *managedIdentityClient) createIMDSAuthRequest(ctx context.Context, clien
 	q := request.URL.Query()
 	q.Add("api-version", c.imdsAPIVersion)
 	q.Add("resource", strings.Join(scopes, " "))
-	if clientID != "" {
-		q.Add(qpClientID, clientID)
+	if c.id == ResourceID {
+		q.Add(string(qpResID), id)
```

Comment:
Unnecessary type-cast?

Reply:
Yes it is! Thanks for catching that, it was a product of my previous implementation where the value of the resource ID was a different type. Done.